tests/cpp_integration_tests/test_main_model.cpp

```diff
@@ -1346,94 +1346,121 @@ TEST_CASE("Test main model - runtime dispatch") {
+                MutableDataset sym_output_from_columnar{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
+                sym_output_from_columnar.add_buffer("node", node_output_from_columnar.size(),
+                                                    node_output_from_columnar.size(), nullptr,
+                                                    node_output_from_columnar.data());
+
+                base_model.calculate(options, sym_output_from_base);
+                row_based_model.calculate(options, sym_output_from_row_based);
+                columnar_model.calculate(options, sym_output_from_columnar);
+
+                REQUIRE(node_output_from_columnar.size() == node_output_from_base.size());
+                REQUIRE(node_output_from_columnar.size() == node_output_from_row_based.size());
+
+                for (Idx idx = 0; idx < std::ssize(node_output_from_columnar); ++idx) {
+                    // check columnar updates work same way as row-based updates
+                    CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_row_based[idx].id));
+                    CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_row_based[idx].u_pu));
+                    // check update actually changed something
+                    CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_base[idx].id));
+                    if (idx == 0) { // sym_load node
+                        CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_base[idx].u_pu));
+                    } else {
+                        CHECK(node_output_from_columnar[idx].u_pu != doctest::Approx(node_output_from_base[idx].u_pu));
+                    }
+                }
+            }
 
-            for (Idx idx = 0; idx < std::ssize(node_output_from_columnar); ++idx) {
-                // check columnar updates work same way as row-based updates
-                CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_row_based[idx].id));
-                CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_row_based[idx].u_pu));
-                // check update actually changed something
-                CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_base[idx].id));
-                if (idx == 0) { // sym_load node
-                    CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_base[idx].u_pu));
-                } else {
-                    CHECK(node_output_from_columnar[idx].u_pu != doctest::Approx(node_output_from_base[idx].u_pu));
+            SUBCASE("Without IDs") {
+                ConstDataset update_data_with_rows{false, 1, "update", meta_data::meta_data_gen::meta_data};
+                update_data_with_rows.add_buffer("sym_load", state.sym_load_update.size(), state.sym_load_update.size(),
```

Comment:
These two lines are not used in this test. However, I believe that row based updates without IDs are not tested yet, are they? If not, please add it (We may want to rename the outter test name).

Reply:
Row based without IDs are not supported atm. From use case point of view, columnar data without ids is almost always the case.

Reply:
Shouldn't it be according to 7. ii. b. b. of #548 ?

Reply:
According to the feature request it is. It can be added later. The PR is about columnar update data after all.

Reply:
Two lines removed.

Reply:
Update: in theory, the code in place should handle row-based update data without ID buffer. I have requested help from @nitbharambe to create such a test to see if those support are indeed working.

Reply:
Update to the update: I put logic in place to prevent this from being used. In the core, the logic could in theory handle, but for now it is not supported. Tried it from the Python side and got the expected error (expecting columnar data)

Reply:
As discussed, conversation resolved.